modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,158 @@
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the `oc apply` command:
++
+[IMPORTANT]
+====
+You must have cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: share-rhel-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a `ClusterRole` to grant permission to access the `SharedSecret` object and a `RoleBinding` object that grants permission to access the `SharedSecret` object by running the following command:
++
+[source,terminal]
+----
+$ oc new-project buildah-entitlement
+----
++
+[source,terminal,subs="attributes+"]
+----
+$ oc apply -f -<<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: use-shared-secret
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - share-entitlement-secret
+    verbs:
+      - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-shared-secret
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: use-shared-secret
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+EOF
+----
+
+. Create a build with relevant mount details using `buildah` buildStrategy:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel-entitlement-build
+spec:
+  source:
+    type: Git
+    git:
+      url: <LOCATION_OF_OPERATOR_ORGANIZATION>
+      revision: quick-tests
+    contextDir: rh-builds
+  strategy:
+    name: buildah-csi
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: dockerfile <1>
+    value: Dockerfile
+  volumes:
+  - csi:
+      driver: csi.sharedresource.openshift.io
+      readOnly: true <2>
+      volumeAttributes:
+        sharedSecret: share-entitlement-secret <3>
+    name: rhel-entitlement
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/buildah-entitlement/sample-go-app
+EOF
+----
++
+[NOTE]
+====
+The output image is saved to the internal registry. The location of the image will change based on the namespace the build runs in.
+====
+
+. Create the buildRun referencing the rhel-entitlement-build:
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: BuildRun
+metadata:
+  name: buildah-rhel-entitlement-buildrun
+spec:
+  build:
+    name: buildah-rhel-entitlement-build
+EOF
+----
++
+[IMPORTANT]
+====
+* You must include the command to remove the `/etc/rhsm-host` directory and all its contents in the Dockerfile before executing any `yum` or `dnf` commands.
```

Comment:
Could we include an example Dockerfile that shows this? As a novice user, I find it difficult to follow what I should do by just reading the docs but this example shows me exactly wheat I need to do:
https://github.com/apoorvajagtap/shipwright-sample-go/blob/main/rh-builds/Dockerfile